redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-details/RejsonDetails.tsx

```diff
@@ -32,8 +32,14 @@ const RejsonDetails = (props: BaseProps) => {
 
   const dispatch = useDispatch()
 
-  const handleFetchVisualisationResults = (path: string, forceRetrieve = false) =>
+  const handleFetchVisualisationResults = (path: string, forceRetrieve = false) => new Promise((resolve, reject) => {
     dispatch<any>(fetchVisualisationResults(path, forceRetrieve))
+      .then((data: any) => resolve({
+        ...data,
+        data: parseJsonData(data?.data)
+      }))
+      .catch(reject)
+  })
```

Comment:
I would suggest parse json in the fetchVisualisationResults

Reply:
`fetchVisualisationResults` do not store results. It's ok to transform result of BE and return prepared data.

Reply:
I see what you mean. Updated it.